runtime/src/bank.rs

```diff
@@ -5180,9 +5237,9 @@ impl Bank {
             &self.rc.accounts.accounts_db,
             &self.ancestors,
             sanitized_txs,
-            check_results,
+            check_results.to_owned(),
```

Comment:
This clones the `check_results` which we were not doing before. Why did this become necessary?
I think cloning these should be *relatively* light, but it does involve an Arc clone.

Reply:
I think we can just pass check results as an owned vector and avoid this clone

Reply:
a slice was suggested by clippy over a vector. in addition filter_executable_program_accounts called from this method modifies the check_results.